gdx/src/com/badlogic/gdx/graphics/g3d/utils/Debug3dRenderer.java

```diff
@@ -0,0 +1,220 @@
+package com.badlogic.gdx.graphics.g3d.utils;
+
+import com.badlogic.gdx.graphics.Camera;
+import com.badlogic.gdx.graphics.Color;
+import com.badlogic.gdx.graphics.GL20;
+import com.badlogic.gdx.graphics.Mesh;
+import com.badlogic.gdx.graphics.VertexAttribute;
+import com.badlogic.gdx.graphics.VertexAttributes.Usage;
+import com.badlogic.gdx.graphics.g3d.Material;
+import com.badlogic.gdx.graphics.g3d.Renderable;
+import com.badlogic.gdx.graphics.g3d.RenderableProvider;
+import com.badlogic.gdx.math.Vector3;
+import com.badlogic.gdx.math.collision.BoundingBox;
+import com.badlogic.gdx.utils.Array;
+import com.badlogic.gdx.utils.Disposable;
+import com.badlogic.gdx.utils.GdxRuntimeException;
+import com.badlogic.gdx.utils.Pool;
+
+/** @author realitix */
+public class Debug3dRenderer implements Disposable, RenderableProvider {
+	private float vertices[];
+	private short indices[];
+	private int idVertice;
+	private int idIndice;
+	private Vector3 tmp = new Vector3();
+	private Vector3 tmp2 = new Vector3();
+	private Renderable renderable = new Renderable();
+	private boolean building;
+	private Color color0 = new Color();
+	private Color color1 = new Color();
+	private Color color2 = new Color();
+	private Color color3 = new Color();
+	private Color color4 = new Color();
+
+	public static int maxVertices = 5000;
+	public static int maxIndices = 5000;
+
+	public Debug3dRenderer () {
+		// Init mesh
+		Mesh mesh = new Mesh(false, maxVertices, maxIndices, new VertexAttribute(Usage.Position, 3, "a_position"),
+			new VertexAttribute(Usage.ColorPacked, 4, "a_color"));
+		// Init vertices
+		vertices = new float[maxVertices * (mesh.getVertexSize() / 4)];
+		// Init indices
+		indices = new short[maxIndices];
+
+		// Init renderable
+		renderable.meshPart.mesh = mesh;
+		renderable.meshPart.offset = 0;
+		renderable.meshPart.primitiveType = GL20.GL_LINES;
+		renderable.material = new Material();
+	}
+
+	public void begin () {
+		if (building) throw new GdxRuntimeException("Call end() after calling begin()");
+		building = true;
+
+		idVertice = 0;
+		idIndice = 0;
+	}
+	
+	public void add (Camera camera) {
```

Comment:
I have a feeling that it should be able to accept a `Frustum` directly and rely on the camera (so a signature `add(Frustum frustum)` should be possible and this one could call `add(camera.frustum);`.